runtime/jvmti/jvmtiMemory.c

```diff
@@ -80,4 +80,21 @@ jvmtiDeallocate(jvmtiEnv* env,
 	TRACE_JVMTI_RETURN(jvmtiDeallocate);
 }
 
-
+jvmtiError JNICALL 
+jvmtiSetHeapSamplingInterval(jvmtiEnv *env, 
+	jint samplingInterval)
+{
+	jvmtiError rc = JVMTI_ERROR_NONE;
+	J9VMThread *currentThread = NULL;
+	
+	Trc_JVMTI_jvmtiSetHeapSamplingInterval_Entry(env, samplingInterval);
+	
```

Comment:
need to ensure that this is called during the OnLoad or the live phase

Reply:
need to check that sampling interval is non-negative

Reply:
what happens if this is called and the JVM doesn't posses the capability to do heap sampling?

Reply:
Added `ENSURE_PHASE_ONLOAD_OR_LIVE`, `ENSURE_CAPABILITY` and `ENSURE_NON_NEGATIVE` accordingly.